File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l = 0, r = heights.length - 1, max = 0
        while(l<r){
            if(heights[l] < heights[r]){
                max = Math.max(max, heights[l]*(r-l))
                l++
            } else {
                max = Math.max(max, heights[r]*(r-l))
                r--
            }
        }
        return max
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
import math

class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:

        def canEat(eatingRate):
            currSum = 0            
            for p in piles:
                currSum += math.ceil(p/eatingRate)
            return currSum <= h

          
        l = 1
        r = max(piles)
        res = r
        while l<=r:
            mid = l + (r-l)//2
            if canEat(mid):
                res = mid
                r = mid - 1
            else:
                l = mid + 1

        return res 
